controller/src/main/java/io/pravega/controller/fault/ControllerClusterListener.java

```diff
@@ -126,24 +120,79 @@ protected void startUp() throws InterruptedException, Exception {
                 }
             };
 
-            taskSweeper.sweepOrphanedTasks(processes);
+            sweepEventProcessorReaders(processes);
+            sweepTransactions(processes);
+            sweepTasks(processes);
+
+            log.info("Controller cluster listener startUp complete");
+        } finally {
+            LoggerHelpers.traceLeave(log, objectId, "startUp", traceId);
+        }
+    }
+
+    private void handleHostRemoved(Host host) {
+        taskSweeper.sweepOrphanedTasks(host.getHostId());
+
+        eventProcessorsOpt.ifPresent(controllerEventProcessors -> {
```

Comment:
Why is this optional? Do we ever run in production without this set? If not and it's for testing we should switch to using mocks so that is not affecting our flow.